components/Form/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,199 @@
+import { useState, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import ErrorMessage from 'components/ErrorMessage/ErrorMessage';
+import DeleteIcon from 'components/Icons/TimesCircle';
+
+import { Controller } from 'react-hook-form';
+import style from './Autocomplete.module.scss';
+
+const AutocompleteComponents = ({
+  name,
+  value,
+  onChange,
+  options,
+  placeholder,
+  label,
+  hint,
+  register,
+  rules,
+  error,
+  labelSize = 'm',
+  required,
+  ...otherProps
+}) => {
+  const data = options.map((option) => {
+    return typeof option === 'string'
+      ? { value: option, text: option }
+      : option;
+  });
+  const [selected, setSelected] = useState(value || '');
+  const [suggestions, setSuggestions] = useState(data);
+  const [suggestionView, setSuggestionView] = useState(false);
+  const [displayView, setDisplayValue] = useState('');
+  const [clearBtn, setClearBtn] = useState(false);
+
+  useEffect(() => {
+    document.addEventListener('mousedown', viewHandler);
+    return () => {
+      document.removeEventListener('mousedown', viewHandler);
+    };
+  }, []);
+
+  const getSuggestion = (value) => {
+    setDisplayValue(value);
+    setClearBtn(true);
+    setSuggestions(
+      data.filter((option) =>
+        option.text.toLowerCase().includes(value.toLowerCase())
+      )
+    );
+  };
+
+  const viewHandler = (e) => {
+    if (e.target.parentElement.id === name || e.target.name === name) {
+      setSuggestionView(true);
+      return;
+    }
+    setSuggestionView(false);
+  };
+
+  const clearValue = () => {
+    setDisplayValue('');
+    setSuggestions(data);
+    setClearBtn(false);
+  };
+  const handleSelected = useCallback((e) => {
+    setSelected(e.target.value);
+    setClearBtn(true);
+    onChange(e.target.value);
+    setDisplayValue(e.target.innerText);
+    setSuggestionView(false);
+  });
+  return (
+    <div className="govuk-!-margin-top-5">
+      <div className={style.inputContainer}>
+        <div
+          className={cx('govuk-form-group', {
+            'govuk-form-group--error': error,
+          })}
+        >
+          <label
+            className={`govuk-label govuk-label--${labelSize}`}
+            htmlFor={name}
+          >
+            {label} {required && <span className="govuk-required">*</span>}
+          </label>
+          {hint && (
+            <span id={`${name}-hint`} className="govuk-hint">
+              {hint}
+            </span>
+          )}
+          {error && <ErrorMessage label={error.message} />}
+          <div className={style.inputContainer}>
+            <input
+              className={style.input}
+              id={name}
+              placeholder={placeholder || 'Search or select a value'}
+              data-testid={name}
+              name={name}
+              value={displayView}
+              defaultValue={selected}
```

Comment:
You shouldn't set `value` and `defaultValue` at the same time, react is not happy about it https://reactjs.org/docs/forms.html#controlled-components